src/app/views/common/copy/KeyboardCopyEvent.ts

```diff
@@ -0,0 +1,40 @@
+import { eventTypes, telemetry } from '../../../../telemetry';
+import { KEYBOARD_COPY_TABS } from '../../../../telemetry/component-names';
+
+interface IComponentList {
+  [key: string]: string;
+}
+export const KeyboardCopyEvent = () => {
+  const componentList: IComponentList = KEYBOARD_COPY_TABS;
+  document.addEventListener('keydown', (event: KeyboardEvent) => {
+    if (event && (event.ctrlKey || event.metaKey) && event.key === 'c') {
+      const targets = event.composedPath();
+      const componentName = getComponentName(targets);
+      trackCopyEvent(componentName);
+    }
+  });
+
+  const getComponentName = (targets: EventTarget[]): string => {
+    const targetIds = targets.map((target: EventTarget) => {
+      return getTargetId(target);
+    });
+    const filteredTargetIds = targetIds.filter((value) => value !== null)!;
+    const componentName = Object.keys(componentList).find(key => filteredTargetIds.includes(key));
+    return componentName || '';
+  }
+
+  const getTargetId = (target: EventTarget) => {
```

Comment:
Instead of id's could we use the itemKey property? 
The other telemetry events use itemKey and I think it's better to maintain consistency... 
I can see a potential issue where one telemetry has the id as `Javascript tab` and another will have `JavaScript tab` which will appear as two different components, yet they are the same component

Reply:
We can't use the itemKey property because this is specific to clicks on pivot items. The global event does not know about this property, so the only other option is to use the target IDs. 

Nice catch. Let me change the spelling

Reply:
Alright awesome